src/groundlight/client.py

```diff
@@ -172,11 +173,62 @@ def list_image_queries(self, page: int = 1, page_size: int = 10) -> PaginatedIma
             image_queries.results = [self._fixup_image_query(iq) for iq in image_queries.results]
         return image_queries
 
+    def ask_confident(
+        self,
+        detector: Union[Detector, str],
+        image: Union[str, bytes, Image.Image, BytesIO, BufferedReader, np.ndarray],
+        wait: Optional[float] = None,
+    ) -> ImageQuery:
+        """Evaluates an image with Groundlight, waiting until an answer above the confidence threshold of the detector is reached or the wait period has passed.
+        :param detector: the Detector object, or string id of a detector like `det_12345`
+        :param image: The image, in several possible formats:
+          - filename (string) of a jpeg file
+          - byte array or BytesIO or BufferedReader with jpeg bytes
+          - numpy array with values 0-255 and dimensions (H,W,3) in BGR order
+            (Note OpenCV uses BGR not RGB. `img[:, :, ::-1]` will reverse the channels)
+          - PIL Image
+          Any binary format must be JPEG-encoded already.  Any pixel format will get
+          converted to JPEG at high quality before sending to service.
+        :param wait: How long to wait (in seconds) for a confident answer.
+        """
+        return self.submit_image_query(
+            detector,
+            image,
+            wait=wait,
+        )
+
+    def ask_fast(
```

Comment:
I think I still prefer `ask_ml` here.

Reply:
The logic fundamentally looks for the first answer we get back, which is why I like ask_fast. 99.9% of the time it is de facto asking ML because the ML returns faster than anything else. I only worry that this could set up some confusion down the road if we have to explain to people why the first result is (almost) the same as the ML result.

I won't dig in too much on this though, it's updated to ask_ml in the latest version now.